instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -905,6 +908,58 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
             .first()
     }
 
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+    @Test
+    fun routeRefreshMultipleAttemptsWhenAlternativesChange() = sdkTest {
+        val refreshStates = mutableListOf<RouteRefreshStateResult>()
+        val statesObserver = RouteRefreshStatesObserver {
+            refreshStates.add(it)
+        }
+        mapboxNavigation.routeRefreshController.registerRouteRefreshStateObserver(statesObserver)
+
+        val routeOptions = generateRouteOptions(twoCoordinates)
+
+        mockWebServerRule.requestHandlers.remove(failByRequestRouteRefreshResponse)
```

Comment:
why do you remove `failByRequestRouteRefreshResponse`? 🤔

Reply:
I want to replace refresh handler. For that I need to remove the old one.